app/models/concerns/foreman_remote_execution/exportable.rb

```diff
@@ -0,0 +1,54 @@
+# This concern makes it easy to export an ActiveRecord object with specified
+# attributes and associations only.  If a specified assocation also includes
+# this concern, then it will likewise be exported. Custom attributes can be
+# specified as an options hast consisting of a custom export lambda.
+#
+# Example:
+#   attr_exportable :name, :address, :company => ->(user) { user.company.name }
+#
+module ForemanRemoteExecution
+  module Exportable
+    extend ActiveSupport::Concern
```

Comment:
>  Would there be any use case in other parts of Foreman for this?

It's reusable for sure, but I can't think of any other uses in core at the moment.  

I also tried to find a gem like you mentioned or something that does this, and couldn't. Rails has `as_json` which can take some options but it didn't handle everything I needed.

> We are using mainly the rabl to do this job

Hmm, that's an interesting idea.  Not RABL, but  we could use a view to render the job template export, which might be simpler than this Exportable concern.


```erb
<%%\#
kind: job_template
name: <%= @template.name %>
template_inputs: <%= template_inputs.each do |input|
  <% render :partial => 'template_inputs'
<% end %>

etc.

%%>
```